core/domain/value_generators_domain_test.py

```diff
@@ -48,6 +61,63 @@ def test_generate_value_of_base_value_generator_raises_error(self) -> None:
                 'generate_value() method has not yet been implemented')):
             base_generator.generate_value()
 
+    def test_registry_template_random_selector_contents(self) -> None:
+        contents_registry = (
+            '<schema-based-editor [schema]="$ctrl.SCHEMA" '
+            'ng-model="$ctrl.customizationArgs.list_of_values">\n'
+            '</schema-based-editor>\n'
+        )
+        class_object = value_generators_domain.Registry()
+        self.assertEqual(
+            contents_registry,
+            class_object.get_generator_class_by_id(
+                'RandomSelector'
+            ).get_html_template()
+        )
+
+    def test_registry_template_copier_contents(self) -> None:
+        contents_registry = (
+            '<span class="d-inline-block align-middle">\n  '
+            '<object-editor obj-type="<[objType]>" init-args="initArgs" '
+            'value="customizationArgs.value" always-editable="true">\n  '
+            '</object-editor>\n</span>\n'
+        )
+        class_object = value_generators_domain.Registry()
+        self.assertEqual(
+            contents_registry,
+            class_object.get_generator_class_by_id(
+                'Copier'
+            ).get_html_template()
+        )
+
+    def test_get_value_generator_classes_not_subclass(self) -> None:
+        """Test  that the value generator registry discovers all classes
+        correctly and excludes classes that are not subclasses of
+        BaseValueGenerator.
+        """
+
+        class MockCopier():
+            """This is a dummy class for self.swap to test  that the value
+            generator registry discovers all classes correctly and excludes
+            classes that are not subclasses of BaseValueGenerator.
+            We need to have a class in the returned list of value generators
+            that isn't a subclass of BaseValueGenerator to test.
+            """
+
+            pass
+
+        module = importlib.import_module(
+            'extensions.value_generators.models.generators'
+        )
+        expected_generators = {
+            'RandomSelector': type(generators.RandomSelector())
+            }
```

Comment:
Deindent by 4 to match opening {.

Reply:
Done

Reply:
Er, not done?

Please push your code changes and only then reply to the comments, so that the code is consistent with your comments. (In general, reviewers get notified once comments are replied to, so by that time the code should be in a reviewable state.) Thanks!

Reply:
Sorry about that. I went back to my VM and it was hung after the type checks finished but never pushed the changes. I pushed again and checked the branch on github and it looks good now.